pages/index.tsx

```diff
@@ -147,69 +174,108 @@ export default function Home({
                   <div className="text-red-500">Lahjansaaja on pakollinen</div>
                 )}
               </div>
-              <Button type="submit" className="mt-8">
+              <Button
+                type="submit"
+                className={`mt-8 ${giftQuery.isFetching || giftQuery.isError ? 'cursor-not-allowed bg-red-500' : null}`}
+                disabled={
+                  giftQuery.isFetching || giftQuery.isError ? true : false
+                }
+              >
                 Lisää
               </Button>
             </form>
           </div>
-          {giftData.length > 0 && (
-            <div className="mt-7">
-              <TitleText className="text-start text-xl">Lahjaideat</TitleText>
-              {giftData.map((giftItem) => (
-                <div
-                  key={`${giftItem.uuid}_divbutton`}
-                  className="mt-4 animate-opacity"
-                >
-                  <div key={giftItem.uuid} className="grid">
-                    <p
-                      className={`hover-target col-start-1 text-primaryText [overflow-wrap:anywhere]`}
-                    >
-                      {giftItem.gift} <span>-</span> {giftItem.receiver}
-                    </p>
-                    <SvgPencilEdit
-                      key={`${giftItem.uuid}_editbutton`}
-                      width={24}
-                      height={24}
-                      className="trigger-underline col-start-2 row-start-1 mr-8 justify-self-end align-middle text-stone-600 hover:cursor-pointer"
-                      onClick={() => {
-                        setEditModalGiftData(giftItem);
-                        setIsEditModalOpen(true);
-                      }}
-                    />
+          <TitleText className="mt-7 text-start text-xl">Lahjaideat</TitleText>
+          <GiftList
+            giftQuery={giftQuery}
+            giftData={giftData}
+            refreshGiftList={() => void refreshGiftList()}
```

Comment:
Tätä parametria ei enää tarvita (ks. toinen kommentti), mutta jos tarvittaisiin se tulisi antaa näin.
```suggestion
            refreshGiftList={refreshGiftList}
```